src/server/svr_jobfunc.c

```diff
@@ -1470,6 +1471,176 @@ svr_chkque(job *pjob, pbs_queue *pque, char *hostname, int mtype)
 	return (0);	/* all ok, job can enter queue */
 }
 
+/**
+ * @brief
+ *		check_block_wt	-	A worktask to reply to the blocked job client
+ *
+ * @param[in]	ptask	-	work_task structure
+ */
+void
+check_block_wt(struct work_task *ptask)
+{
+	struct block_job_reply *blockj = ptask->wt_parm1;
+	int time_difference;
+	time_t current_timestamp = time(NULL);
+	int is_connected = 0;
+	int is_replied = 0;
+	int is_error = 0;
+
+	if (blockj->ji_block_fd == -1) {
+		int sock_flags;
+		int	is_connect;
+		struct hostent		*hp;
+		struct sockaddr_in	remote;
+		short			remote_sin_family;
+
+		if ((hp = gethostbyname(blockj->ji_submit_host)) == NULL) {
+			sprintf(log_buffer, "%s: client host %s not found for block job %s", __func__,
+			blockj->ji_submit_host, blockj->ji_jobid);
+			log_event(PBSEVENT_ERROR, PBS_EVENTCLASS_JOB, LOG_NOTICE,
+				blockj->ji_jobid, log_buffer);
+			free(blockj->ji_msg);
+			free(blockj);
+			return;
+		}
+
+		remote_sin_family = hp->h_addrtype;
+		memset(&remote, 0, sizeof(remote));
+		memcpy(&remote.sin_addr, hp->h_addr, hp->h_length);
+		remote.sin_port = htons((unsigned short)blockj->ji_port);
+		remote.sin_family = remote_sin_family;
+
+		blockj->ji_block_fd = socket(AF_INET, SOCK_STREAM, 0);
+		/* Set socket to Non-blocking */
+		if (blockj->ji_block_fd != -1) {
+			sock_flags = fcntl(blockj->ji_block_fd, F_GETFL, 0);
+			if (fcntl(blockj->ji_block_fd, F_SETFL, sock_flags | O_NONBLOCK) == -1) {
+				close(blockj->ji_block_fd);
+				blockj->ji_block_fd = -1;
+			}
+		}
+
+		/* The flag "is_connected" indicates the readiness of the socket and could be one of the following:
```

Comment:
Just a suggestion: For ease of code readability!
Could we rename this flag as "**is_ready_to_write**"? and the flag values in the sequence order 0) Socket not created 1) Connect() in Progress 2) Successful connect() and poll(); Socket is ready to write msg.

Reply:
Hi @bremanandjk, thank you for the suggestion. I agree with renaming the flag. I will make the changes. Regarding the sequence order, I kept it to be 1 for `is_connected` which indicates boolean for success(i.e. socket is ready for write). Also, in a case where I am polling and upon success, I make the variable from 2 to 1 again. I would keep the flag values as it is.